front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts

```diff
@@ -195,28 +218,51 @@ export class CreateF3XStep1Component extends FormComponent implements OnInit {
     }
 
     const summary: Form3X = Form3X.fromJSON(SchemaUtils.getFormValues(this.form, f3xSchema, this.formProperties));
+    summary.filing_frequency = this.form.get('filing_frequency')?.value;
+    summary.report_type_category = this.form.get('report_type_category')?.value;
 
     // If a termination report, set the form_type appropriately.
     if (summary.report_code === ReportCodes.TER) {
       summary.form_type = F3xFormTypes.F3XT;
     }
 
-    const create$ = this.form3XService.create(summary, this.formProperties);
+    let report: Report;
+    if (this.report?.id) {
+      summary.id = this.report?.id;
+      try {
+        report = await this.form3XService.update(summary, this.formProperties, [HttpStatusCode.BadRequest]);
+      } catch {
+        this.coverageDatesDialogVisible = true;
+        return;
+      }
+    } else {
+      report = await this.form3XService.create(summary, this.formProperties);
+    }
 
-    //Create the report
-    create$.then((report) => {
+    if (report) {
       if (jump === 'continue') {
         this.router.navigateByUrl(`/reports/transactions/report/${report.id}/list`);
       } else {
-        this.router.navigateByUrl('/reports');
+        if (this.report) {
```

Comment:
I think it still needs to go to reports if it exists.

Reply:
added this back.